Add sample data

tychuang1211 opened the pull request:
This PR adds the following:
1. A sample project containing default datasets in various CKAN-previewable formats (CSV, JSON, GeoJSON, SHP, XML, KML, etc.)
~~2. A default test user account~~
~~3. A default private dataset~~

The dataset content may still be adjusted later, but this version can already be used to test whether it work as expected.

tychuang1211 committed:
Add sample data

tychuang1211 commented:
@u10313335 I'm still coordinating with Ally on the dataset content, but it's ready for functional testing. I'd appreciate any feedback on whether it works as expected.

u10313335 commented on ckan/sample-data/data/sample_data.html:
There is an internal server error when previewing this source.

u10313335 commented on ckan/docker-entrypoint.d/04_setup_sample_data.sh:
```diff
@@ -0,0 +1,13 @@
+
+cd /srv/app/sample-data
+
+ckan user add demo email=demo@localhost password=demo1234
```
Do not create a new user. Use the `ckan_admin` user defined in the `.env` instead.

u10313335 commented on ckan/sample-data/datasets.jsonl:
```diff
@@ -0,0 +1,5 @@
+{"author":"內政部國土測繪中心","author_email":null,"contact_email":"","contact_person":"","created_time":"2015-10-07","creator_user_id":"1c54b003-eef7-43a9-b6d6-46cb3fba2271","data_type":["science"],"groups":[],"isopen":true,"keywords":[],"language":["zho"],"license_id":"twogd","license_title":"TWOGDL","license_url":"https://data.gov.tw/license","maintainer":null,"maintainer_email":null,"metadata_created" ...
+{"author":"臺北市政府工務局水利工程處","author_email":null,"contact_email":"","contact_person":"","created_time":"2016-06-30","creator_user_id":"1c54b003-eef7-43a9-b6d6-46cb3fba2271","data_type":["science"],"groups":[],"isopen":true,"keywords":[],"language":[],"license_id":"twogd","license_title":"TWOGDL","license_url":"https://data.gov.tw/license","maintainer":null,"maintainer_email":null,"metadata_created": ...
+{"author":"The depositar lab","author_email":null,"contact_email":"data.contact@depositar.io","contact_person":"The depositar team | 研究資料寄存所計畫團隊","creator_user_id":"1c54b003-eef7-43a9-b6d6-46cb3fba2271","data_type":["doc","image","multimedia"],"groups":[],"isopen":true,"keywords":["Q98433008"],"language":["eng"],"license_id":"cc-by","license_title":"CC-BY 4.0","license_url":"https://creativecommo ...
+{"author":"None","author_email":null,"contact_email":"","contact_person":"","creator_user_id":"1c54b003-eef7-43a9-b6d6-46cb3fba2271","data_type":["other"],"groups":[],"isopen":false,"keywords":[],"language":[],"license_id":"notspecified","license_title":"License Not Specified","maintainer":null,"maintainer_email":null,"metadata_created":"2025-06-09T06:48:43.807950","metadata_modified":"2025-06-09 ...
```
```suggestion
```

I would suggest that all sample datasets be public and and for demonstration purposes only, so there is no need to create another account — the sample datasets can be created directly by the sysadmin account (`ckan_admin`).

u10313335 commented on ckan/sample-data/upload-files.jsonl:
```diff
@@ -0,0 +1,9 @@
+{"action":"resource_patch","data":{"id":"51ca2530-408c-4234-b558-5baf054cdf3f"},"files":{"upload":"./data/sample_data.json"}}
+{"action":"resource_patch","data":{"id":"851f6f13-73db-407c-8cba-7ad1bf453b12"},"files":{"upload":"./data/sample_data.kml"}}
+{"action":"resource_patch","data":{"id":"57e3da70-c7b1-4cb5-98fd-a528b4853b36"},"files":{"upload":"./data/sample_data.html"}}
+{"action":"resource_patch","data":{"id":"5ac7312e-e071-46ff-9854-0da9aba2c56d"},"files":{"upload":"./data/sample_data.pdf"}}
```
Some filenames here are somewhat inconsistent with the `datasets.jsonl`. Maybe the inconsistency leads to the internal server error?

u10313335 reviewed (changes requested):
Additionally, I'm wondering if it's possible to remove the internal fields such as `id` and `metadata_created` to make the metadata of the sample dataset more concise.

u10313335 commented on kompose/ckan-deployment.yaml:
Was this file added by accident?

tychuang1211 replied:
It's added by accident, I'll remove it.

u10313335 commented:
There are two issues for now:

- Resources in datasets other than `sample-dataset-2` cannot be viewed. It seems that these resources were not successfully uploaded. I have tried `docker compose down -v` and run `docker compose build` again.
- Please add a base layer for map widgets (used by the spatial search and previews):
  ```diff
  diff --git a/.env.example b/.env.example
  index ede74b5..8ec87d8 100644
  --- a/.env.example
  +++ b/.env.example
  @@ -73,3 +73,8 @@ CKAN__HARVEST__MQ__TYPE=redis
   CKAN__HARVEST__MQ__HOSTNAME=redis
   CKAN__HARVEST__MQ__PORT=6379
   CKAN__HARVEST__MQ__REDIS_DB=1
  +
  +# Spatial
  +CKANEXT__SPATIAL__COMMON_MAP__TYPE=custom
  +CKANEXT__SPATIAL__COMMON_MAP__CUSTOM__URL=//tile.openstreetmap.org/{z}/{x}/{y}.png
  +CKANEXT__SPATIAL__COMMON_MAP__ATTRIBUTION=&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors
  ```

tychuang1211 committed:
Remove demo user and private dataset

tychuang1211 committed:
Update `.env.example`

tychuang1211 committed:
Make metadata more concise

tychuang1211 committed:
Remove upload logo

tychuang1211 commented:
> There are two issues for now:
> 
>     * Resources in datasets other than `sample-dataset-2` cannot be viewed. It seems that these resources were not successfully uploaded. I have tried `docker compose down -v` and run `docker compose build` again.
> 
>     * Please add a base layer for map widgets (used by the spatial search and previews):
>       ```diff
>       diff --git a/.env.example b/.env.example
>       index ede74b5..8ec87d8 100644
>       --- a/.env.example
>       +++ b/.env.example
>       @@ -73,3 +73,8 @@ CKAN__HARVEST__MQ__TYPE=redis
>        CKAN__HARVEST__MQ__HOSTNAME=redis
>        CKAN__HARVEST__MQ__PORT=6379
>        CKAN__HARVEST__MQ__REDIS_DB=1
>       +
>       +# Spatial
>       +CKANEXT__SPATIAL__COMMON_MAP__TYPE=custom
>       +CKANEXT__SPATIAL__COMMON_MAP__CUSTOM__URL=//tile.openstreetmap.org/{z}/{x}/{y}.png
>       +CKANEXT__SPATIAL__COMMON_MAP__ATTRIBUTION=&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors
>       ```

I've cleaned up the metadata and updated the .env file.
Could you try rebuilding and check if the issues you mentioned still persist?

u10313335 commented:
> > There are two issues for now:
> > ```
> > * Resources in datasets other than `sample-dataset-2` cannot be viewed. It seems that these resources were not successfully uploaded. I have tried `docker compose down -v` and run `docker compose build` again.
> > 
> > * Please add a base layer for map widgets (used by the spatial search and previews):
> >   ```diff
> >   diff --git a/.env.example b/.env.example
> >   index ede74b5..8ec87d8 100644
> >   --- a/.env.example
> >   +++ b/.env.example
> >   @@ -73,3 +73,8 @@ CKAN__HARVEST__MQ__TYPE=redis
> >    CKAN__HARVEST__MQ__HOSTNAME=redis
> >    CKAN__HARVEST__MQ__PORT=6379
> >    CKAN__HARVEST__MQ__REDIS_DB=1
> >   +
> >   +# Spatial
> >   +CKANEXT__SPATIAL__COMMON_MAP__TYPE=custom
> >   +CKANEXT__SPATIAL__COMMON_MAP__CUSTOM__URL=//tile.openstreetmap.org/{z}/{x}/{y}.png
> >   +CKANEXT__SPATIAL__COMMON_MAP__ATTRIBUTION=&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors
> >   ```
> > ```
> 
> I've cleaned up the metadata and updated the .env file. Could you try rebuilding and check if the issues you mentioned still persist?

The situation continues. And all resources are gone now. I would try to deploy on another machine later.

u10313335 commented on .env.example:
```diff
@@ -73,3 +73,28 @@ CKAN__HARVEST__MQ__TYPE=redis
 CKAN__HARVEST__MQ__PORT=6379
 CKAN__HARVEST__MQ__REDIS_DB=1
+
+## Spatial Settings
```
```suggestion
## Spatial settings
```

u10313335 commented on .env.example:
```diff
@@ -73,3 +73,28 @@ CKAN__HARVEST__MQ__TYPE=redis
+## ckanext-data-depositario Settings
+CKANEXT__DATA_DEPOSITARIO__GMAP__API_KEY =
+CKANEXT__DATA_DEPOSITARIO__DEMO__ENABLED = true
+CKANEXT__DATA_DEPOSITARIO__SITE_NOTICE = 下次清除時間 Next Purge:
```
```suggestion
## ckanext-data-depositario settings
CKANEXT__DATA_DEPOSITARIO__GMAP__API_KEY =
CKANEXT__DATA_DEPOSITARIO__DEMO__ENABLED = true
```
The notice for the next purge time is no longer necessary once auto-rebuild is enabled.

u10313335 commented on .env.example:
```diff
@@ -73,3 +73,28 @@ CKAN__HARVEST__MQ__TYPE=redis
+CKANEXT__CITATION__CSL_MAPPINGS = {"url": "erc_where"}
+
+## ckanext-ark settings
+CKANEXT__ARK__NAAN = 99999
```
```suggestion
```

`ckanext.ark.naan` has been defined in the [custom_options_extra.ini](https://github.com/depositar/ckanext-data-depositario/blob/master/config/custom_options_extra.ini), which is [applied](https://github.com/depositar/depositar-docker/blob/prod-demo/ckan/Dockerfile#L29) during image build.

u10313335 commented on ckan/sample-data/upload-files.jsonl:
```diff
@@ -0,0 +1,8 @@
+{"action":"resource_create","data":{"package_id":"sample-dataset-1","format":"CSV","name":"海巡署臺、澎、金、馬地區各安檢所及海巡隊地址 CSV 格式"},"files":{"upload":"./data/sample_data.csv"}}
+{"action":"resource_create","data":{"package_id":"sample-dataset-1","format":"XML","name":"海巡署臺、澎、金、馬地區各安檢所及海巡隊地址 XML 格式"},"files":{"upload":"./data/sample_data.xml"}}
+{"action":"resource_create","data":{"package_id":"sample-dataset-1","format":"JSON","name":"海巡署臺、澎、金、馬地區各安檢所及海巡隊地址 JSON 格式"},"files":{"upload":"./data/sample_data.json"}}
+{"action":"resource_create","data":{"package_id":"sample-dataset-2","format":"SHP","name":"直轄市、縣市界線 (TWD97 經緯度) SHP 格式"},"files":{"upload":"./data/sample_data.zip"}}
```
```suggestion
{"action":"resource_create","data":{"package_id":"sample-dataset-2","format":"SHP","name":"直轄市、縣市界線 (TWD97 經緯度) SHP 格式","encoding":"utf-8"},"files":{"upload":"./data/sample_data.zip"}}
```

In depositar, the `ckanext.geoview.shp_viewer.encoding` is set to `big5`, which is the encoding used by most existing Shapefile. However, this particular Shapefile is encoded in UTF-8, so its encoding must be specified explicitly.

tychuang1211 committed:
Apply suggestions from code review

Co-authored-by: Sol Lee <u103133.u103135@gmail.com>

u10313335 commented:
> > > There are two issues for now:
> > > ```
> > > * Resources in datasets other than `sample-dataset-2` cannot be viewed. It seems that these resources were not successfully uploaded. I have tried `docker compose down -v` and run `docker compose build` again.
> > > 
> > > * Please add a base layer for map widgets (used by the spatial search and previews):
> > >   ```diff
> > >   diff --git a/.env.example b/.env.example
> > >   index ede74b5..8ec87d8 100644
> > >   --- a/.env.example
> > >   +++ b/.env.example
> > >   @@ -73,3 +73,8 @@ CKAN__HARVEST__MQ__TYPE=redis
> > >    CKAN__HARVEST__MQ__HOSTNAME=redis
> > >    CKAN__HARVEST__MQ__PORT=6379
> > >    CKAN__HARVEST__MQ__REDIS_DB=1
> > >   +
> > >   +# Spatial
> > >   +CKANEXT__SPATIAL__COMMON_MAP__TYPE=custom
> > >   +CKANEXT__SPATIAL__COMMON_MAP__CUSTOM__URL=//tile.openstreetmap.org/{z}/{x}/{y}.png
> > >   +CKANEXT__SPATIAL__COMMON_MAP__ATTRIBUTION=&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors
> > > ```
> > 
> > 
> > I've cleaned up the metadata and updated the .env file. Could you try rebuilding and check if the issues you mentioned still persist?
> 
> The situation continues. And all resources are gone now. I would try to deploy on another machine later.

It turns out that the incompatible [qsvdp](https://github.com/depositar/depositar-docker/blob/prod-demo/ckan/Dockerfile#L35-L38) (the binary used by the [datapusher-plus](https://github.com/dathere/datapusher-plus) extension) leads to the failed uploads. The qsvdp in the repo is compiled for the `x64` platform, which is incompatible with my `aarch64/arm64` macOS. But since the target platform will be x64, the current setup is okay.

u10313335 commented:
~~A further tweak: add the site title to the `.env.example`: `CKAN__SITE_TITLE=depositar-demo` (after the `CKAN_SITE_ID`).~~ I will do it.

u10313335 reviewed (approved):
LGTM
